fuel-tx/src/transaction/types/mint.rs

```diff
@@ -94,6 +73,9 @@ impl crate::UniqueIdentifier for Mint {
         }
 
         let mut clone = self.clone();
+        clone.input_contract.prepare_sign();
+        clone.output_contract.prepare_sign();
+
         crate::transaction::compute_transaction_id(chain_id, &mut clone)
```

Comment:
This is starting to look quite ineffiience, but maybe there's no nice way to not clone here?

Reply:
Agreed, although probably not worth optimizing until we have real benchmarks.

Reply:
We do it for each `Transaction` variant inside the `id` function. We could optimize by directly zeroing bytes in the final buffer, but it could add complexity and bugs to our code, so cloning is okay for now. Plus, we cache the `TxId` in most cases, so it is a one-time operation.